### PR TITLE
Do not block on duplicate base classes

### PR DESCRIPTION
**Pull request by sobolevn**
@JelleZijlstra I don't really have a prefence on this issue. I am fine with both blocking and non-blocking errors here.

But, it looks like that making this non-blocking is quite straight-forward.
Closes https://github.com/python/mypy/issues/15349

### CODE REVIEW

File: test-data/unit/check-classes.test [hunk 1]
```diff
@@ -4131,6 +4131,18 @@ class C(str, str):
...
 
+[case testDupBaseClassesIsNotBlocking]
+class A:
+    def method(self) -> str: ...
+
+class B(A, A):  # E: Duplicate base class "A"
+    ...
+
+b: B
+# We use dummy type info, so no real attrs will be there:
```
Comment (JelleZijlstra) [inline test-data/unit/check-classes.test:4142]: That's pretty unfortunate, can we make it inherit from Any or just skip the duplicate base instead?

Comment (sobolevn) [reply test-data/unit/check-classes.test]: Yeap, got it. Fixed!

File: test-data/unit/check-classes.test [hunk 2]
```diff
@@ -4131,6 +4131,20 @@ class C(str, str):
 [out]
```
Comment (hauntsaninja) [inline test-data/unit/check-classes.test:4131]: nit: maybe we remove testMroSetAfterError? Your new test case seems like a superset

**Review by hauntsaninja — COMMENTED** [review]
This looks good to me!

One possible improvement here could be to deduplicate the base classes when calculating MRO, so instead of Any MRO we still get the MRO as if the base class was unduplicated.

**Comment by sobolevn** [comment]
@hauntsaninja I don't fill confortable doing this. Because we can have two generic bases: `Some[int]` and `Some[str]` - which one we should remove? This way we can generate strange errors that won't help much. `Any` is the default way to handle errors inside, so let's stick to it.

**Comment by hauntsaninja** [comment]
Okay! I went ahead and added a test case for that situation

**Pull request merged**